### PR TITLE
more automation, less mistakes

### PR DESCRIPTION
**Pull request by shiftkey**
- [x] wireup Travis to run against PRs
- [x] see build errors because of files not being marked with `.yml`
- [x] validate a bunch of attributes
- [x] actually exit if an error is found
- [x] catch duplicate tags
- [x] catch malformed tags

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+language: ruby
+rvm:
+  - 2.2
+
+sudo: false
+
+install:
+  - bundle install --path vendor/bundle
```
Comment (akoeplinger) [inline .travis.yml]: add `bundle config build.nokogiri --use-system-libraries` before that one to avoid building nokogiri during build.

Comment (shiftkey) [inline .travis.yml]: :metal: done further down in the file

**Comment by shiftkey** [comment]
Awful ruby code aside, that was easier than I thought it'd be...

**Comment by shiftkey** [comment]
Ready for review, but given this is not part of the core I'm not sure how thorough it needs to be...

**Pull request merged**